tests/channel_test.c

```diff
@@ -206,18 +206,17 @@ static void s_wait_a_bit_task(struct aws_task *task, void *arg, enum aws_task_st
     aws_condition_variable_notify_one(cv);
 }
 
-static int s_wait_a_bit(struct aws_event_loop *loop) {
+static int s_wait_a_bit(struct aws_event_loop *loop, struct aws_task *task) {
     struct aws_mutex mutex = AWS_MUTEX_INIT;
     struct aws_condition_variable cv = AWS_CONDITION_VARIABLE_INIT;
     ASSERT_SUCCESS(aws_mutex_lock(&mutex));
 
-    struct aws_task task;
-    aws_task_init(&task, s_wait_a_bit_task, &cv, "wait_a_bit");
+    aws_task_init(task, s_wait_a_bit_task, &cv, "wait_a_bit");
 
     uint64_t run_at_ns;
     ASSERT_SUCCESS(aws_event_loop_current_clock_time(loop, &run_at_ns));
     run_at_ns += aws_timestamp_convert(1, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_NANOS, NULL);
-    aws_event_loop_schedule_task_future(loop, &task, run_at_ns);
+    aws_event_loop_schedule_task_future(loop, task, run_at_ns);
 
     return aws_condition_variable_wait(&cv, &mutex);
```

Comment:
we talked about this.
intention of helper function is to wait for task-queue to flush
bug is: spurious wake-ups of condition variables could cause the function to exit before task actually executed
simpler implementation would be spin-locking on an atomic bool